### DISCUSSION TITLE
Battery parameters

### DISCUSSION THREAD

**Discussion by bengtner** [wlcrs/huawei_solar#121] [General]
Is there any more info available about how to control battery charging/discharging? 

It seems like my battery is configured to charge as soon as the PV yields power, and discharges at night. However, this is not an optimal setup when selling price (daytime) is 100 times higher than what I pay at night. As long as selling/purchasing  price is unbalanced like this, I should avoid discharging the battery. How can I configure such a model?

**Comment by TheLordFolken** [comment]
Your Battery has been configured for maximum discharge. The alternative settings is "Time of Use" which defines rules for when the battery can be charged or discharged. This is not an integration specific settings but is a setting that can be configured by your installer, via the fusion solar web interface and via this integration. However, you need to have installer level permission.

**Reply by bengtner** [reply]
Ok, I will start to elaborate all options. I have enabled "elevated permissions" (or whatever it is called) in the Modbus over TCP settings. I hope this is enough to enable this setting.

**Comment by wlcrs** [comment]
That is indeed what the "Maximum self consumption" mode does.

The easiest way is to set the 'battery maximum charging power' and 'battery
maximum discharging power' dynamically to suit your needs.
